mijia/src/bluetooth.rs

```diff
@@ -164,7 +166,7 @@ impl BluetoothSession {
                     .filter_map(|(k, v)| {
                         let k = k.as_str()?.into();
                         let v = v.box_clone();
-                        let v = cast::<Vec<u8>>(&v)?.clone();
+                        let v = cast::<Variant<Vec<u8>>>(&v)?.0.clone();
```

Comment:
Pre-approved for when you push the change to use into_iter() etc. I really think that the best time to fix it properly is as part of #9 so I'm happy to leave it as-is for now.

If you want to write a test that's realistic, you will probably need to add an extra layer of nesting, so you can't Any your way out of trouble without boxed_clone, (we're currently a hashmap of {sv}, where the v is a Variant of &[u8] or something, so we're already borrowing from the outer hashmap)